robot_skills/src/robot_skills/arms.py

```diff
@@ -22,6 +22,50 @@
 # Temporary: this should be approximately 0.02 once the sensor is correctly setup
 GRASP_SENSOR_LIMITS = tuple(rospy.get_param("skills/arm/grasp_sensor/limits", [0.0025, 0.18]))
 
+class PublicArm(object):
+    """
+    Public arm interface, also checks a challenge doesn't try to use more than it asked for.
+
+    :ivar _arm: Private link to the real arm.
+    :ivar gripper_type: Gripper type to be used.
+    :ivar joint_goals: Joint goals to be used.
+    :ivar joint_trajectories: Joint trajectories to be used.
+    """
+    def __init__(self, arm, gripper_type, joint_goals, joint_trajectories):
+        self._arm = arm
+        self.gripper_type = gripper_type
+        self.joint_goals = joint_goals
+        self.joint_trajectories = joint_trajectories
+
```

Comment:
Add getter+setter `@property` for what the arm is occupied by. These work with the ID of an Ed Entity